monads/test/Spec.hs

```diff
@@ -1,2 +1,70 @@
+module Main (main) where
+
+import qualified Data.Map as M
+
+import HW.StackMachine
+import HW.Compiler
+import HW.Eval
+import Expr
+
+import Test.Tasty.HUnit
+import Test.Tasty
+
+
 main :: IO ()
-main = putStrLn "Test suite not yet implemented"
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Tests"
+  [ compilerTests
+  , evalTests
+  ]
+
+compilerTests :: TestTree
+compilerTests = testGroup "Compiler Tests"
+  [ testCase "Compile single number" $
+      assertEqual "" [PushNum 5] (compile (Num 5 :: Expr String))
```

Comment:
use `@?=` if you're not going to pass a message to `assertEqual`.

Reply:
[Fixed](https://github.com/kajigor/fp-2024-cub-fall/pull/96/commits/59ba4ea732e81290ab4a7f0a86e7a19c4d0321d4#diff-0e851a1a04a54af327b8044f27cdd1c408a2fee485bb48c4638e5656bdb5e8a3)